subprojects/code-quality/src/main/groovy/org/gradle/api/plugins/quality/CustomizableHTMLReport.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.plugins.quality;
+
+import org.gradle.api.Incubating;
+import org.gradle.api.reporting.SingleFileReport;
+
+/**
+ * A HTML Report whose generation can be customized with a XSLT stylesheet.
+ */
+@Incubating
+public interface CustomizableHTMLReport extends SingleFileReport {
+
+    /**
+     * The stylesheet to use to generate the HTML report.
+     *
+     * @return the stylesheet to use to generate the HTML report
+     */
+    String getStylesheet();
```

Comment:
Please use a [TextResource](https://github.com/gradle/gradle/blob/master/subprojects/core/src/main/groovy/org/gradle/api/resources/TextResource.java) as the type of the stylesheet instead of String. It is a richer datatype and it allows for easier testing. Also, when you do want to get the path to the file, you can always use `getStylesheet().asFile().getAbsolutePath()` for passing it to the findbugs command-line.